### PR TITLE
Port clearsky-based quality checks from Solar Forecast Arbiter

### PR DESCRIPTION
**Pull request by wfvining**
Adds quality test to flag irradiance measurements that exceed the expected clearsky irradiance.

## Changes from SFA
* consolidate POA and GHI checks into a single function. These were separate in SFA, but ultimately do exactly the same thing. For pvanalytics it makes sense to consolidate them into a single general purpose function: `quality.irradiance.clearsky_limits()`.
* remove the `@mask_flags` decorator.

### CODE REVIEW

File: pvanalytics/quality/irradiance.py [hunk 1]
```diff
@@ -309,3 +310,35 @@ def check_irradiance_consistency_qcrad(ghi, solar_zenith, dhi, dni,
                              bounds=bounds['low_zenith']))
 
     return consistent_components, diffuse_ratio_limit
+
+
+def clearsky_limits(measured, clearsky, csi_max=1.1):
+    """Identify irradiance measurements greater than clearsky values.
```
Comment (cwhanse) [inline pvanalytics/quality/irradiance.py:316]: Here, we "identify...greater than"  The function returns True when a value is less than or equal to.  Do we want to return True if greater than, or keep this behavior?

Comment (wfvining) [reply pvanalytics/quality/irradiance.py:316]: I think we want to keep the current behavior. It matches the semantics of the qcrad validation functions in this module which return True for values that are physically plausible.

To make the documentation more clear, perhaps it should say "Identify irradiance measurements that do not exceed clearsky values"

Comment (cwhanse) [inline pvanalytics/quality/irradiance.py:316]:
```suggestion
    """Identify irradiance values which do not exceed clearsky values.
```

File: pvanalytics/quality/irradiance.py [hunk 2]
```diff
@@ -309,3 +310,35 @@ def check_irradiance_consistency_qcrad(ghi, solar_zenith, dhi, dni,
...
+
+    Uses :py:func:`pvlib.irradiance.clearsky_index` to compute the
+    clearsky index for `measured` and `clearsky`. Compares the
```
Comment (cwhanse) [inline pvanalytics/quality/irradiance.py:319]:
```suggestion
    clearsky index as the ratio of `measured` to `clearsky`. Compares the
```

File: pvanalytics/quality/irradiance.py [hunk 3]
```diff
@@ -309,3 +310,35 @@ def check_irradiance_consistency_qcrad(ghi, solar_zenith, dhi, dni,
...
+    clearsky index to `csi_max` to identify values in `measured` that
+    are greater than the expected clearsky irradiance.
```
Comment (cwhanse) [inline pvanalytics/quality/irradiance.py:321]:
```suggestion
    are less than or equal to `csi_max`.
```

**Pull request merged**